sklearn/metrics/cluster/_supervised.py

```diff
@@ -138,6 +146,206 @@ def contingency_matrix(labels_true, labels_pred, *, eps=None, sparse=False):
+
+    D11
+      number of pairs with both clusterings having the samples
+      clustered together
+
+    Parameters
+    ----------
+    labels_true : int array, shape = [n_samples]
+        Ground truth class labels to be used as a reference
+
+    labels_pred : array, shape = [n_samples]
+        Cluster labels to evaluate
+
+    Returns
+    -------
+    nested array [[D00, D01], [D10, D11]]
+
+    Examples
+    --------
+    Perfectly matching labelings have all non-zero entries on the
+    diagonal regardless of actual label values:
+
+      >>> from sklearn.metrics.cluster import pair_confusion_matrix
+      >>> pair_confusion_matrix([0, 0, 1, 1], [0, 0, 1, 1])
+      array([[8, 0],
+             [0, 4]]...
+
+      >>> pair_confusion_matrix([0, 0, 1, 1], [1, 1, 0, 0])
+      array([[8, 0],
+             [0, 4]]...
+
+    Labelings that assign all classes members to the same clusters
+    are complete but may be not always pure, hence penalized, and
+    have some off-diagonal non-zero entries:
+
+      >>> pair_confusion_matrix([0, 0, 1, 2], [0, 0, 1, 1])
+      array([[8, 2],
+             [0, 2]]...
+
+    The matrix is not symmetric:
+
+      >>> pair_confusion_matrix([0, 0, 1, 1], [0, 0, 1, 2])
+      array([[8, 0],
+             [2, 2]]...
+
+    If classes members are completely split across different clusters, the
+    assignment is totally incomplete, hence the matrix has all zero
+    diagonal entries:
+
+      >>> pair_confusion_matrix([0, 0, 0, 0], [0, 1, 2, 3])
+      array([[ 0,  0],
+             [12,  0]]...
+
+    References
+    ----------
+    .. L. Hubert and P. Arabie, Comparing Partitions, Journal of
+      Classification 1985
+      https://link.springer.com/article/10.1007%2FBF01908075
+
+    See also
+    --------
+    adjusted_pair_confusion_matrix: Adjusted Rand Score
+    adjusted_mutual_info_score: Adjusted Mutual Information
+
+    """
+    labels_true, labels_pred = check_clusterings(labels_true, labels_pred)
+    n_samples = np.int64(labels_true.shape[0])
+
+    # Computation using the contingency data
+    contingency = contingency_matrix(labels_true, labels_pred, sparse=True,
+                                     dtype=np.int64)
+    sum_comb = sum(_comb2(n_ij) for n_ij in contingency.data)
+    sum_comb_c = sum(_comb2(n_c) for n_c in np.ravel(contingency.sum(axis=1)))
+    sum_comb_k = sum(_comb2(n_k) for n_k in np.ravel(contingency.sum(axis=0)))
+    D11 = 2 * sum_comb
+    D01 = 2 * (sum_comb_k - sum_comb)
+    D10 = 2 * (sum_comb_c - sum_comb)
+    D00 = n_samples * n_samples - n_samples - D01 - D10 - D11
+    return np.array([[D00, D01], [D10, D11]])
+
+
+def rand_score(labels_true, labels_pred):
+    """Rand index.
+
+    The Rand Index computes a similarity measure between two clusterings
+    by considering all pairs of samples and counting pairs that are
+    assigned in the same or different clusters in the predicted and
+    true clusterings.
+
+    The raw RI score is:
```

Comment:
Similarly remove this and reference the User Guide

Reply:
Done.